solutions/usaco-380.mdx

```diff
@@ -131,5 +131,118 @@ public class CrossCountrySkiing {
 
 </JavaSection>
 
+<CPPSection>
+
+```cpp
+// @author: ascyrax
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define endl "\n"
+
+#define pb push_back
+
+int r, c;
+vector<vector<int>> grid;     //grid stores the elevation
+vector<vector<bool>> visited; //this will be true if the corresponding location of the grid has been visited atleast once
+
+//right bottom left top
+int drow[4] = {0, 1, 0, -1};
+int dcol[4] = {1, 0, -1, 0};
+vector<pair<int, int>> wpts; //this will contain all the waypoint
+
+
+void dfs(int row, int col, int mid)
+{
+        // cout << row << " " << col << " " << mid << endl;
+        if (row < 0 || col >= c)
+        {
+                return;
+        }
+        visited[row][col] = true;
+        for (int i = 0; i <= 3; i++)
+        {
+                int newRow = row + drow[i];
+                int newCol = col + dcol[i];
+                if ((newRow >= 0 && newRow < r && newCol >= 0 && newCol < c) && !visited[newRow][newCol] && (abs(grid[newRow][newCol] - grid[row][col]) <= mid))
+                        dfs(newRow, newCol, mid);
+        }
+        return;
+}
+
+bool check()
+{
+        for (auto i : wpts)
+        {
+                if (visited[i.first][i.second] == false) //checking if all the waypoints were visited in the last dfs, or not
```

Comment:
line too long here as well

Reply:
also add a space after the `//` for all comments

Reply:
Ok. Will keep this in mind now onwards.